cap-primitives/src/winx/fs/dir_utils.rs

```diff
@@ -35,8 +36,14 @@ pub(crate) fn append_dir_suffix(path: PathBuf) -> PathBuf {
 // Strip trailing `/`s, unless this reduces `path` to `/` itself. This is
 // used by `mkdir` and others to prevent paths like `foo/` from canonicalizing
 // to `foo/.` since these syscalls treat these differently.
-pub(crate) fn strip_dir_suffix(path: &Path) -> &Path {
-    unimplemented!("strip_dir_suffix")
+pub(crate) fn strip_dir_suffix(path: &Path) -> Cow<Path> {
```

Comment:
Or if you want, it looks like you could change the result type to `impl Deref<Target = Path> + '_`, which would allow it to be the same on Windows and Unix.

Reply:
Sure, I wanted to avoid different return types so that we always return the same type across all platforms. IMHO this should aid readability and maintainability. How about we make an unconditional alloc here and return `PathBuf` on all platforms instead?

Reply:
`impl Deref` sounds great!

Reply:
Comment race :o). What do you think of the suggestion above?

Reply:
I like the `impl Deref` approach. Lemme try it out!

Reply:
`impl Deref` it is!